evaluate/metrics/assessment.go

```diff
@@ -57,6 +63,24 @@ func (a Assessments) Add(x Assessments) {
 	}
 }
 
+// IsEqual checks if both assessment collections are equal.
+func (a Assessments) IsEqual(x Assessments) bool {
+	if a == nil {
```

Comment:
Better not allocate, its wasteful. Add a 

if len(x) != len(y) { return false }

at the top. handles the nil cases right away.

Reply:
IsEqual -> Equal

Like in assert.Equal.

Reply:
> if len(x) != len(y) { return false }

But that one is incorrect if one assessment explicitly sets a key to `0` and the other one does not set it, meaning we have the default value `0` so they end up being equal. (https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1570206152)

Reply:
Changed it to:

```go
if a == nil || x == nil {
	return a == nil && x == nil
}
```

Reply:
Ah i see